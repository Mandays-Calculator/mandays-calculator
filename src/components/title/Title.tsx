import type { ReactElement } from "react";
import { styled } from "@mui/material/styles";
import Typography from "@mui/material/Typography";

interface TitleProps {
  title: string;
  color?: string;
}

const StyledTitle = styled(Typography)`
  font-size: 2.5rem;
`;

const Title = (props: TitleProps): ReactElement => {
  const { title, color = "primary" } = props;
  return (
    <StyledTitle color={color} fontWeight="600">
      {title}
    </StyledTitle>
  );
};

export default Title;
