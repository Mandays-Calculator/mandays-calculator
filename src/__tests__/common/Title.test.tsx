import { render } from "@testing-library/react";
import Title from "~/components/title/Title";

describe("Title Component", () => {
  it("renders the title component", () => {
    const { getByTestId, asFragment } = render(
      <Title
        title="Title Component"
        color="primary"
      />
    );

    expect(getByTestId("title-component")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
