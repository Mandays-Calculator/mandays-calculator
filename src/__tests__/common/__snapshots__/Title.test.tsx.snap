// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component renders the title component 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-g6ng2y-MuiTypography-root"
    data-testid="title-component"
  >
    Title Component
  </p>
</DocumentFragment>
`;
